js/setup.js

```diff
@@ -4,11 +4,133 @@ var WIZARD_NAMES = ['Иван', 'Хуан Себастьян', 'Мария', 'К
+var setup = document.querySelector('.setup');
+var setupOpen = document.querySelector('.setup-open');
+var setupClose = document.querySelector('.setup-close');
+
+// нахожу волшебника
+var playerWizard = document.querySelector('.setup-player');
+// все инпуты и визуальные элементы волшебника - глаза, плащ, фаерболл
+var inputPlayerWizardCoat = document.querySelector('input[name="coat-color"]');
+var playerWizardCoat = playerWizard.querySelector('.wizard-coat');
+var inputPlayerWizardEyes = document.querySelector('input[name="eyes-color"]');
+var playerWizardEyes = playerWizard.querySelector('.wizard-eyes');
+var inputPlayerWizardFireball = document.querySelector('input[name="fireball-color"]');
+var playerWizardFireball = document.querySelector('.setup-fireball-wrap');
+
+// функция, которая по клику на конкретный элемент будет:
+var onElementClick = function (evt) {
+  if (evt.target.matches('.wizard-eyes')) {
+    var eyesColor = getRandomProperty(WIZARD_EYES_COLORS); // 1) вызывать функция определения случайного элемента массива и сохранять в переменную
+    playerWizardEyes.style.fill = eyesColor; // 2) окрашивать элемент разметки цветом, определенным в переменной
+    inputPlayerWizardEyes.value = eyesColor; // 3) передавать значение этой же переменной в value инпута, чтобы передать на сервер
+  } else if (evt.target.matches('.wizard-coat')) {
+    var coatColor = getRandomProperty(WIZARD_COAT_COLORS);
+    playerWizardCoat.style.fill = coatColor;
+    inputPlayerWizardCoat.value = coatColor;
+  } else if (evt.target.matches('.setup-fireball')) {
+    var fireballColor = getRandomProperty(WIZARD_FIREBALL_COLORS);
+    playerWizardFireball.style.backgroundColor = fireballColor;
+    inputPlayerWizardFireball.value = fireballColor;
+  }
+};
+
+// получаю поле ввода имени персонажа
+var userNameInput = document.querySelector('.setup-user-name');
+var isFocus = false; // флаг, который будет проверять, если ли фокус у элемента
+
+// функции, которые будут переключить флаг (может зря я их было вынесла в отдельные, а нужно было сделать просто коллбэками внутри слушателей?)
+var onFocusElement = function () {
+  isFocus = true;
+};
+var onBlurElement = function () {
+  isFocus = false;
+};
+// функция, которая по нажатию на клавишу будет
+var onPopupEscPress = function (evt) {
+  if (evt.keyCode === ESCAPE_KEY_CODE && isFocus) { // если эта клавишу Escape и на поле ввыода имени есть фокус(фраг = true), то ничего не делать
+    return;
+  } else if (evt.keyCode === ESCAPE_KEY_CODE) { // если эта клавишу Escape и на поле ввыода имени нет фокуса, то будет закрывать попап
+    evt.preventDefault();
+    closePopup();
+  }
+};
+
+// функция, при вызове которой будет удаляться класс hidden, добавляться первый слушатель события на документ и второй слушатель события на пользовательского волшебника
+var openPopup = function () {
+  setup.classList.remove('hidden');
+  userNameInput.addEventListener('focus', onFocusElement);
```

Comment:
```suggestion
  userNameInput.addEventListener('focus', onUserNameInputFocus);
```
https://up.htmlacademy.ru/javascript/20/criteries#d4
> on + (на каком элементе) + что случилось:

ПС: я граммар-наци, особенно, что касается названий обработчиков - это моё любимое :) А смысл такой: как только ты называешь обработчики по правилам и ни шагу в сторону, то ты обнаружишь, как много может быть у тебя одинаковых слушателей на одном элементе, но тк они назывались по разному, ты это не видела, а если, есть то, что ты не видишь в своем коде, значит, ты кодом не управляешь. Особенно это показательно у document.